libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFPasscodeViewController.m

```diff
@@ -22,180 +22,162 @@
  WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#import <QuartzCore/QuartzCore.h>
-#import <objc/runtime.h>
 #import "SFPasscodeViewController.h"
 #import "SFPasscodeManager.h"
 #import "SFSDKResourceUtils.h"
+#import "SFSDKPasscodeTextField.h"
+#import <LocalAuthentication/LocalAuthentication.h>
+#import "SFKeychainItemWrapper.h"
+#import "SFSDKWindowManager.h"
+#import "SFSDKPasscodeViewConfig.h"
 
 // Private view layout constants
-static CGFloat      const kPaddingTop                       = 25.0f;
-static NSString *   const kPasscodeTextFontName             = @"HelveticaNeue-Bold";
-static CGFloat      const kPasscodeTextFontSize             = 29.0f;
-static NSString *   const kPasscodeHelperTextFontName       = @"HelveticaNeue";
-static CGFloat      const kPasscodeHelperTextFontSize       = 13.0f;
-static NSUInteger   const kMaxPasscodeLength                = 8;
-static CGFloat      const kControlPadding                   = 5.0f;
-static CGFloat      const kTextFieldWidthPadding            = 40.0f;
-static CGFloat      const kErrorLabelHeight                 = 35.0f;
-static CGFloat      const kInstructionsLabelHeight          = 50.0f;
-static CGFloat      const kLabelPadding                     = 10.0f;
-static CGFloat      const kForgotPasscodeButtonWidth        = 150.0f;
-static CGFloat      const kForgotPasscodeButtonHeight       = 40.0f;
-static CGFloat      const kUseTouchIdButtonWidth            = 150.0f;
-static CGFloat      const kUseTouchIdButtonHeight           = 40.0f;
-
-@interface SFPasscodeViewController() {
+static NSUInteger   const kMaxPasscodeLength                 = 8;
+static CGFloat      const kDefaultPadding                    = 20.0f;
+static CGFloat      const kIconCircleDiameter                = 80.0f;
+static CGFloat      const kTopPadding                        = 64.5f;
+static CGFloat      const kFaceIdIconWidth                   = 36.0f;
+static CGFloat      const kTouchIdIconWidth                  = 42.0f;
+static CGFloat      const kTouchIconPadding                  = 19.0f;
+static CGFloat      const kFaceIconPadding                   = 22.0f;
+static CGFloat      const kTitleTextLabelHeight              = 20.0f;
+static CGFloat      const kPassodeInstructionsLabelHeight    = 20.0f;
+static CGFloat      const kBiometricInstructionsLabelHeight  = 40.0f;
+static CGFloat      const kButtonCornerRadius                = 4.0f;
+static CGFloat      const kButtonHeight                      = 47.0f;
+static CGFloat      const kVerifyButtonWidth                 = 143.0f;
+static CGFloat      const kPasscodeViewHeight                = 48.0f;
+static CGFloat      const kViewBoarderWidth                  = 1.0f;
+
```

Comment:
I feel like the SFPasscodeViewController has become too complex to understand. We should consider a refactor or simple redesign in the long run. There's too much going on within the view controller for managing all the various view components and other internal state. I think we should consider refactoring the code into  more modular views (SetupPasscodeView, VerifyPassCodeView, BioMetricView) and their delegates.  Lot of the view state can hence be contained into the views themselves. It should make it easier to extend and maintain. @wmathurin @bhariharan what do you think?